File: services/user-service/src/main/java/info/novatec/spring/showcase/user/api/resource/UserResource.java
package info.novatec.spring.showcase.user.api.resource;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.springframework.hateoas.ResourceSupport;

import java.util.UUID;

@Data
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
public class UserResource extends ResourceSupport {

  private UUID identifier;

  private Long version;

  private String firstName;

  private String lastName;

  private String displayName;

  private boolean registered;
}

File: webapp/src/app/resources/user/UserResource.ts
import { ResourceLink } from "../ResourceLink";

export class UserResource {
  public identifier: string;
  public version: number;
  public displayName: string;
  public firstName: string;
  public lastName: string;
  public registered: boolean;

  public _links: UserLinks;
}

class UserLinks {
  public profilePictureUrl: ResourceLink;
}